src/codec/builtin/SqlPageCodec.ts

```diff
@@ -48,55 +48,55 @@ export class SqlPageCodec {
 
         // Read column types.
         const columnTypeIds: number[] = ListIntegerCodec.decode(clientMessage);
-        const columnTypes: SqlColumnType[] = [];
+        const columnTypes: SqlColumnType[] = new Array<SqlColumnType>(columnTypeIds.length);
 
         // Read columns
-        const columns: any[][] = [];
+        const columns: any[][] = new Array(columnTypeIds.length);
 
-        for (const columnTypeId of columnTypeIds) {
+        for (const [i, columnTypeId] of columnTypeIds.entries()) {
```

Comment:
can you replace the for .. of loops with the standard one + length caching in hotpaths such as this? It is significantly faster 

[see this](https://perf.link/#eyJpZCI6ImJibjNtbTA0bHJkIiwidGl0bGUiOiJpdGVyIiwiYmVmb3JlIjoiY29uc3QgZGF0YSA9IFsuLi5BcnJheSgxMDAwKS5rZXlzKCldIiwidGVzdHMiOlt7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiY29uc3QgbGVuID0gZGF0YS5sZW5ndGg7XG5mb3IgKGxldCBpID0gMDsgaSA8IGxlbjsgaSsrKSB7XG4gIGNvbnN0IGQgPSBkYXRhW2ldO1xufSIsInJ1bnMiOlsxMzkzNTU1LDEzMzUwNTUsMTM2MzM4OCwxMTM0NjExLDEzMTcyNzcsMTM3ODY2NiwxMzI5MzMzLDExNTE0NDQsMTM4MzY2NiwxMTMxNjExLDEyODEwMDAsMTM0MDU1NSwxMzQ1Mjc3LDEyOTQzMzMsMTM5NjYxMSwxMzY4Mjc3LDE0MDMyNzcsMTI5MjUwMCwxMzg5Nzc3LDEzODgxNjYsMTM2Mjc3NywxMzMwMTY2LDEzNjAwNTUsMTM1MjIyMiwxMzY5NzIyLDEzODg4MzMsMTM4NTc3NywxMzg5MDAwLDEzNjEzMzMsMTM3Nzc3Nyw4NDExNjYsMTMwMDgzMywxMzEzMTExLDEzODMyMjIsMTMxNzExMSwxNDA1MTExLDEzMjY1NTUsMTQwODE2NiwxMzc4NTU1LDEzMzk3MjIsMTM1MzI3NywxMzgyNTU1LDEzNzA3MjIsMTM4MTU1NSwxMzU2Mzg4LDE0MDI1MDAsMTM1NTMzMywxMzIyNDQ0LDEzNzcwMDAsMTQwMDI3NywxMzY2MzMzLDE0MjMyMjIsMTM1MDg4OCwxMzYyMDAwLDEzOTc5NDQsMTQwODUwMCwxMzgyNzIyLDEzMzU2MTEsMTM5NDY2NiwxMzY3NjExLDEzOTU0NDQsMTMwNzUwMCwxNDA4MzMzLDEyNjM0NDQsMTMzNzcyMiwxMjgzMTExLDEzMzEwNTUsMTMwNDIyMiwxMzQ0MDU1LDEzNjE1MDAsMTM2NTExMSwxMzY2NjExLDE0MTAxNjYsMTMxNzE2NiwxMzcyODMzLDEzNjQwNTUsMTM3OTQ0NCwxMzQ5NTAwLDEzNTg4ODgsMTM4Mzg4OCwxMzgwMTExLDEzNzE2MTEsMTM0MzUwMCwxMzk3OTQ0LDEzMjQzODgsMTQxMTYxMSwxNDAwNzc3LDEzNTAwNTUsMTQxNTc3NywxMzQ3Nzc3LDEzMzYyMjIsMTM3OTgzMywxMzY2MTExLDEzNTgwNTUsMTM3MDY2NiwxMzU4NTU1LDExNjQyMjIsMTQxNzU1NSwxMzI4MTExLDEzNzU2MTFdLCJvcHMiOjEzNDgwMzZ9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJmb3IgKGNvbnN0IFtpLCBkXSBvZiBkYXRhLmVudHJpZXMoKSkge1xufSIsInJ1bnMiOls2MDMzMyw1Mjc3Nyw1NDMzMyw0NDk0NCw1NjY2Niw1OTQ0NCw1ODUwMCw1MzU1NSw1NjQ0NCw1NjI3Nyw0OTAwMCw2MDIyMiw1ODQ0NCw1NDE2Niw2MTYxMSw1NjM4OCw1NjcyMiw1MzE2Niw1NjExMSw2NDIyMiw1ODUwMCw2MDY2Niw1NDk0NCw1MzcyMiw1NTIyMiw1NjcyMiw2MjQ0NCw2MjU1NSw1NjAwMCw1MzE2NiwxMzMzMyw1NzI3Nyw0OTUwMCw1NDU1NSw2MDIyMiw1ODQ0NCw1MzY2Niw2MTQ0NCw1MzE2Niw1MzA1NSw1NzUwMCw1ODYxMSw1NjY2Niw1NjcyMiw2MjAwMCw2MTAwMCw2MzUwMCw1NTk0NCw1NTY2Niw1NjExMSw1NTM4OCw2NTUwMCw1NDk0NCw2MTk0NCw1OTAwMCw2MDI3Nyw0OTY2Niw2Mjc3Nyw1OTYxMSw2Mzg4OCw2MTE2Niw1NDk0NCw2MDU1NSw0Nzg4OCw2MDIyMiw2MTYxMSw1MjgzMyw2MTMzMyw1NTY2Niw2MDExMSw2NTE2Niw2MzIyMiw1ODg4OCw1NjYxMSw1NzExMSw0NjQ0NCw1NzE2Niw1NzI3Nyw1NTM4OCw1ODM4OCw1OTE2Niw1MzExMSw1NTA1NSw1NDc3Nyw1NjQ0NCw2MjMzMyw1NjY2Niw2MDYxMSw2MjQ0NCw2MDcyMiw1NDIyMiw1OTAwMCw0OTgzMyw1NDg4OCw1NTA1NSw2MzMzMyw1MTM4OCw2MTE2Niw2MDc3Nyw1NTc3N10sIm9wcyI6NTY4NzR9XSwidXBkYXRlZCI6IjIwMjEtMDUtMzFUMDc6MjU6NTUuMjA3WiJ9)